server/controllers/converters/locationDetailsConverter.ts

```diff
@@ -3,38 +3,64 @@ import {
-    isTemporaryLocation: locationDetails.isTemporaryLocation,
-    movedIn: fromDateTime && formatDateTime(fromDateTime, 'short', ' - '),
-    movedOut: endDateTime && formatDateTime(endDateTime, 'short', ' - '),
-    movedInBy: locationDetails.movementMadeByStaffDetails
-      ? formatName(staffDetails.firstName, '', staffDetails.lastName)
-      : locationDetails.movementMadeByUsername,
+export function locationDetailsConverter(
+  prisonerNumber: string,
+  defaultLocationHistoryToDate: Date = new Date(),
+): (location: LocationDetails) => LocationDetailsForDisplay {
+  return location => {
+    if (!location || !prisonerNumber) return null
+
+    const staffDetails = location.movementMadeByStaffDetails
+    const fromDateTime = location.assignmentDateTime
+    const endDateTime = location.assignmentEndDateTime
+
+    return {
+      establishment: location.agencyName,
+      location: location.location,
+      movedIn: fromDateTime && formatDateTime(fromDateTime, 'short', ' - '),
+      movedOut: endDateTime && formatDateTime(endDateTime, 'short', ' - '),
+      movedInBy: location.movementMadeByStaffDetails
+        ? formatName(staffDetails.firstName, '', staffDetails.lastName)
+        : location.movementMadeByUsername,
+      locationHistoryLink: generateLocationHistoryLink(location, prisonerNumber, defaultLocationHistoryToDate),
+    }
   }
 }
 
-export function convertGroupedLocationDetails(
-  groupedLocationDetails: LocationDetailsGroupedByPeriodAtAgency,
-): GroupedLocationDetailsForDisplay {
-  if (!groupedLocationDetails) return null
-  return {
-    agencyName: groupedLocationDetails.agencyName,
-    fromDate: formatDate(groupedLocationDetails.fromDate, 'short'),
-    toDate: formatDate(groupedLocationDetails.toDate, 'short') || 'Unknown',
-    locationDetails: groupedLocationDetails.locationDetails.map(convertLocationDetails),
+export function groupedLocationDetailsConverter(
+  prisonerNumber: string,
+  defaultLocationHistoryToDate: Date = new Date(),
+): (groupedLocations: LocationDetailsGroupedByPeriodAtAgency) => GroupedLocationDetailsForDisplay {
+  return (groupedLocations: LocationDetailsGroupedByPeriodAtAgency) => {
+    if (!groupedLocations) return null
+    const convertLocationDetails = locationDetailsConverter(prisonerNumber, defaultLocationHistoryToDate)
+    return {
+      agencyName: groupedLocations.agencyName,
+      fromDate: formatDate(groupedLocations.fromDate, 'short'),
+      toDate: formatDate(groupedLocations.toDate, 'short') || 'Unknown',
+      locationDetails: groupedLocations.locationDetails.map(convertLocationDetails),
+    }
   }
 }
+
+function generateLocationHistoryLink(
+  location: LocationDetails,
+  prisonerNumber: string,
+  defaultLocationHistoryToDate: Date,
+): string {
+  if (location.isTemporaryLocation || !location.assignmentDateTime) return null
+
+  const locationHistoryLink = new URL(`/prisoner/${prisonerNumber}/location-history`, config.serviceUrls.digitalPrison)
```

Comment:
Should this not be the location history page in the profile service?
Or is that page not ready to "go live" yet?